notebooks/ImvizExample.ipynb

```diff
@@ -92,69 +102,61 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "acs_47tuc_1 = download_file('https://mast.stsci.edu/api/v0.1/Download/file?uri=mast:HST/product/jbqf03gjq_flc.fits', cache=True)\n",
-    "acs_47tuc_2 = download_file('https://mast.stsci.edu/api/v0.1/Download/file?uri=mast:HST/product/jbqf03h1q_flc.fits', cache=True)"
+    "files = ['jw02731-o001_t017_nircam_clear-f090w_i2d.fits',\n",
+    "         #'jw02731-o001_t017_nircam_clear-f187n_i2d.fits',\n",
+    "         #'jw02731-o001_t017_nircam_clear-f200w_i2d.fits',\n",
+    "         #'jw02731-o001_t017_nircam_clear-f335m_i2d.fits',\n",
+    "         #'jw02731-o001_t017_nircam_clear-f444w_i2d.fits',\n",
+    "         'jw02731-o001_t017_nircam_f444w-f470n_i2d.fits'\n",
+    "        ]\n",
+    "for fn in files:\n",
+    "    uri = f\"mast:JWST/product/{fn}\"\n",
+    "    result = Observations.download_file(uri, base_url=\"https://mast.stsci.edu/api/v0.1/Download/file\")\n",
```

Comment:
Maybe more of a question for @havok2063 -- Is there no way to have MAST cache this somewhere outside of the source checkout tree or at least let user provide the cache directory? Currently, this downloads the big files into the notebooks directory, which poses two different risks:

1. If I do a `git clean -xdf` (which I do from time to time), I will have to redownload these again when I next run the notebook.
2. If someone accidentally added these to version control (you can overwrite `.gitignore` ignore if you are not careful though not easy but also not impossible), we will have to do some crazy git-fu to squash it out of git history (especially if it is not caught early).

Reply:
Yes, you can specify a location of the download path with `local_path`, and you can set a `cache` boolean flag.  If you are downloading files from the production domain, i.e.`mast.stsci.edu`, then you do not need to specify the `base_url` keyword.  See https://astroquery.readthedocs.io/en/latest/api/astroquery.mast.ObservationsClass.html#astroquery.mast.ObservationsClass.download_file.

Reply:
Thanks! In that case, I propose we add a `local_path=Path.home() / 'mastDownloads'` input to the `Observations.download_file()` calls.

Reply:
@havok2063 It seems that local_path expects a filename, not a directory, contrary to the documentation. Should I report this as an issue in `astroquery`?

```
 [Failed]

WARNING: Found cached file /Users/rosteen/MAST_Downloads with size 64 that is different from expected size 603515520 [astroquery.query]

---------------------------------------------------------------------------
IsADirectoryError                         Traceback (most recent call last)
Cell In [4], line 11
      9 for fn in files:
     10     uri = f"mast:JWST/product/{fn}"
---> 11     result = Observations.download_file(uri, local_path=Path.home() / 'MAST_Downloads/', cache=True)
     12     with warnings.catch_warnings():
     13         warnings.simplefilter('ignore')

File ~/opt/anaconda3/envs/viz_dev/lib/python3.10/site-packages/astroquery/mast/observations.py:568, in ObservationsClass.download_file(self, uri, local_path, base_url, cache, cloud_only)
    565             self._download_file(data_url, local_path,
    566                                 cache=cache, head_safe=True, continuation=False)
    567 else:
--> 568     self._download_file(data_url, local_path,
    569                         cache=cache, head_safe=True, continuation=False)
    571 # check if file exists also this is where would perform md5,
    572 # and also check the filesize if the database reliably reported file sizes
    573 if (not os.path.isfile(local_path)) and (status != "SKIPPED"):

File ~/opt/anaconda3/envs/viz_dev/lib/python3.10/site-packages/astroquery/query.py:437, in BaseQuery._download_file(self, url, local_filepath, timeout, auth, continuation, cache, method, head_safe, **kwargs)
    433     progress_stream = io.StringIO()
    435 with ProgressBarOrSpinner(length, f'Downloading URL {url} to {local_filepath} ...',
    436                           file=progress_stream) as pb:
--> 437     with open(local_filepath, open_mode) as f:
    438         for block in response.iter_content(blocksize):
    439             f.write(block)

IsADirectoryError: [Errno 21] Is a directory: '/Users/rosteen/MAST_Downloads'
```

Reply:
whoops, I missed this.  Yeah I would report this as a bug in astroquery.